stdlib/public/core/Array.swift

```diff
@@ -1835,14 +1838,17 @@ extension Array {
   /// and enums.
   ///
   /// The following example copies the bytes of the `numbers` array into a
-  /// buffer of `UInt8`:
+  /// buffer of `UInt8`; on a [little-endian][1] machine, the result is as
+  /// shown:
   ///
-  ///     var numbers = [1, 2, 3]
+  ///     var numbers: [Int32] = [1, 2, 3]
```

Comment:
Why does this example need a type annotation to be added?

Reply:
This was my suggestion, so that the issue of word size does not come up. The same has already been done by someone else for the other example that is edited in this PR. Introducing the issue of platform word size is irrelevant to understanding the functionality of the API here, so we should avoid it if we can, and we can.